### PR TITLE
Handle 403 errors in descriptions and openers

### PR DESCRIPTION
**Pull request by jmorel**
When trying to access a description or an opener for which you had no permission, the frontend was failing silently.

This PR adds a paragraph that is displayed when the backend returned a 403 error while trying to fetch a description or an opener.

Since the description saga was the same everywhere, I mutualized it into the common sagas file.

In the dataset tabs, I had an issue with the detection of empty trainDataSampleKeys lists, I added a double check.

### CODE REVIEW

File: src/app/business/common/components/detail/components/forbiddenDescription.js [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+const Span = styled('span')`
+    margin-right: ${spacingNormal};
+`;
+
+const ForbiddenDescription = ({model, setTabIndex, permissionsTabIndex}) => (
+    <Fragment>
+        <Span>
+            {`You do not have enough permissions to see this ${model}'s description.`}
+        </Span>
+        <RoundedButton onClick={() => setTabIndex(permissionsTabIndex)}>
```
Comment (GuillaumeCisco) [inline src/app/business/common/components/detail/components/forbiddenDescription.js]: this is not the best way to handle the onClick.
With this, on each render, a new function will be created, allocating new javascript memory.
We should better use a class component here for putting this function in hoisting.

File: src/app/business/routes/dataset/components/detail/components/tabs/index.js [hunk 2]
```diff
@@ -45,11 +52,26 @@ descLoading, item, tabIndex, openerLoading, setTabIndex, addNotification,
...
+                    setTabIndex={setTabIndex}
+                />
+            )}
+            {!descLoading && !descForbidden && <Description item={item} />}
         </TabPanel>
         <TabPanel>
             {openerLoading && <PulseLoader size={6} />}
-            {!openerLoading && item && item.opener && item.opener.content && (
+            {!openerLoading && openerForbidden && (
+                <Fragment>
```
Comment (GuillaumeCisco) [inline src/app/business/routes/dataset/components/detail/components/tabs/index.js]: Should it be in its own component as `ForbiddenDescription` is?
Should we create a more generic `Fordidden` component?

File: src/app/business/common/components/detail/components/forbiddenDescription.js (hunk 1 above)
Comment (jmorel) [reply src/app/business/common/components/detail/components/forbiddenDescription.js]: You're absolutely right, I went too fast here. I'm fixing it right away!

File: src/app/business/routes/dataset/components/detail/components/tabs/index.js (hunk 2 above)
Comment (jmorel) [reply src/app/business/routes/dataset/components/detail/components/tabs/index.js]: This makes a lot of sense, I'll update it!

File: src/app/business/common/components/detail/components/forbiddenDescription.js (hunk 1 above)
Comment (jmorel) [reply src/app/business/common/components/detail/components/forbiddenDescription.js]: Done!

File: src/app/business/routes/dataset/components/detail/components/tabs/index.js (hunk 2 above)
Comment (jmorel) [reply src/app/business/routes/dataset/components/detail/components/tabs/index.js]: Done!

**Pull request merged**